doc/concepts/configuration.rst

```diff
@@ -0,0 +1,613 @@
+..  _configuration:
+
+Configuration
+=============
+
+There are two approaches to configuring Tarantool:
+
+*   *Since version 3.0*: In a YAML file.
+
+    In a YAML file, you can provide the full cluster topology and specify all configuration options.
+    You can also use :ref:`etcd <configuration_centralized>` to store configuration data in one reliable place.
+
+*   *In version 2.11 and earlier*: :ref:`In code <configuration_code>` using the ``box.cfg`` API.
+
+    In this case, configuration is provided in a Lua initialization script.
+    Starting with the 3.0 version, configuring Tarantool in code is considered a legacy approach.
+
+
+..  _configuration_overview:
+
+Configuration overview
+----------------------
+
+A YAML configuration file describes the full topology of a Tarantool cluster.
+A cluster's topology includes the following elements, starting from the lower level:
+
+-   An *instance* is a member of the cluster that stores data or might act as a router for handling CRUD requests in a :ref:`sharded <sharding>` cluster.
+-   A *replica set* is a pack of instances that operate on copies of the same databases.
+    :ref:`Replication <replication>` provides redundancy and increases data availability.
+-   A *group* provides the ability to organize replica sets.
+    For example, in a sharded cluster, one group can contain :ref:`storage <vshard-architecture-storage>` instances and another group can contain :ref:`routers <vshard-architecture-router>` used to handle CRUD requests.
+
+You can flexibly configure a cluster's settings on different levels: from global settings applied to all groups to parameters specific for concrete instances.
+
+
+..  _configuration_file:
+
+Configuration in a file
+~~~~~~~~~~~~~~~~~~~~~~~
+
+This section provides an overview on how to configure Tarantool in a YAML file.
+
+.. _configuration_instance_basic:
+
+Basic instance configuration
+****************************
+
+The example below shows a sample configuration of a single Tarantool instance:
+
+..  literalinclude:: /code_snippets/snippets/config/instances.enabled/instance_scope/config.yaml
+    :language: yaml
+    :dedent:
+
+-   The ``instances`` section includes only one instance named *instance001*.
+    The ``iproto.listen`` option sets a port used to listen for incoming requests.
+-   The ``replicasets`` section contains one replica set named *replicaset001*.
+-   The ``groups`` section contains one group named *group001*.
```

Comment:
It seems, we should mention validation rules for instance/replicaset/group names somewhere. See https://github.com/tarantool/tarantool/blob/10372d0d9ded18ff64c256e125ce740bb0605587/src/box/lua/config/cluster_config.lua#L120-L141.

Reply:
Hmm... I think, such info should be placed in the `Configuration reference` (which is not ready yet). Will add your note to the corresponding ticket.